### ISSUE TITLE
🐛[BUG] BetaSchemaForm 使用  request 获取数据, 会导致 formRef 的方法丢失

### ISSUE THREAD

**Issue by mxsl-gr** [ant-design/pro-components#7026]
### 🐛 bug 描述

使用 BetaSchemaForm 的 **request** 获取数据, 会导致 FormInstance 的方法丢失, 获取 formRef.current 是一个空 Object.
但是如果通过 initialValues 赋值则没有此问题.

### 📷 复现步骤

使用 BetaSchemaForm 的 request 获取数据, 并通过 onValuesChange 监听触发 formInstance 的 setFieldsValue.
通过输入框修改, 会触发: `formRef.current?.setFieldsValue is not a function`

### 🏞 期望结果

在使用 request 获取数据时, 能正常获取 ref 的方法

### 💻 复现代码

https://codesandbox.io/s/antd-pro-form-ref-bug-j8y5un?file=/App.tsx

或者核心代码:

```TypeScript
import type {
  ProFormColumnsType,
  ProFormInstance
} from "@ant-design/pro-components";
import { BetaSchemaForm } from "@ant-design/pro-components";
import React from "react";

type DataItem = {
  title: string;
};

const columns: ProFormColumnsType<DataItem>[] = [{
  title: "标题",
  dataIndex: "title"
}];

export default () => {
  const formRef = React.useRef<ProFormInstance<DataItem>>();
  return (
    <BetaSchemaForm<DataItem>
      formRef={formRef}
      request={() => ({
        title: "蚂蚁设计有限公司"
      })}
      onValuesChange={(_, values) => {
        formRef.current?.setFieldsValue({ title: values.title + "!" });
      }}
      columns={columns}
    />
  );
};
```

### © 版本信息

- ProComponents 版本: 2.4.12

**Comment by mxsl-gr** [comment]
发现两周了, 提的 bug issue 一点回应都没有, 大概率这个项目后期活跃度会很低了, 问了一下集团的同学, 蚂蚁内部目前不使用这套东西...后期我也考虑整体替换掉...

**Comment by mxsl-gr** [comment]
> ModalForm同样有问题，貌似跟这个改动有关系 [06721c3#diff-d86203524813bab4cca2d473195acbabce09a751687e117fa19deaa076053a9bR118](https://github.com/ant-design/pro-components/commit/06721c310adde229ce5bcf92bf3ce59100e713df#diff-d86203524813bab4cca2d473195acbabce09a751687e117fa19deaa076053a9bR118)
> 
> 这里附上test code
> 
> ```
>   it('📦 ModelForm get formRef when use request', async () => {
>     const formRef = createRef<FormInstance>();
>     const wrapper = render(
>         <ModalForm
>             open
>             formRef={formRef}
>             request={async (params) => {
>               return params;
>             }}
>             params={{
>               name: 'test',
>             }}
>         >
>           <ProFormText name="name"/>
>         </ModalForm>,
>     );
>     await waitTime(100);
>     expect(formRef.current?.getFieldValue("name")).toBe("test");
>     wrapper.unmount();
>   });
> ```
> 
> @chenshuai2144

临时处理的话, 可以自己初始化一个 state, 获取数据之后通过 initialValues 目前可以临时绕过这个 Bug

**Comment by Seven777Z** [comment]
StepsForm的formMapRef也有此问题

**Comment by chenshuai2144** [comment]
好啦，最近孩子出生，时间少了很多。hhh

**Comment by mxsl-gr** [comment]
> 好啦，最近孩子出生，时间少了很多。hhh

小孩确实会很吃精力和时间, 哈哈哈

**Comment by JackRoy1992** [comment]
啥时候修复了，@下我

**Comment by foureyeschao** [comment]
> @JackRoy1992 2.5.0修了
没看到啊？？？
@ant-design/pro-components@2.5.0
2023-05-29

fix(layout): 修复 collapsedShowTitle 样式错误问题 (#7098). [#7098](https://github.com/ant-design/pro-components/pull/#7098) [@ONLY-yours](https://github.com/ONLY-yours)
fix(form): 解决 ProFormList 多重嵌套 ProFormList 下的表单组件未调用 transform 方法问题 (#7138). [#7138](https://github.com/ant-design/pro-components/pull/#7138) [@kaneruan](https://github.com/kaneruan)
fix(form): use oninit reset formRef. [1deedf2](https://github.com/ant-design/pro-components/commit/1deedf2)
fix(form): close icon style (#7091). [#7091](https://github.com/ant-design/pro-components/pull/#7091) [@kungege](https://github.com/kungege)
fix(list): Extend ProListMeta<T> type with renderFormItem ([#7136](https://github.com/ant-design/pro-components/pull/#7136)). #7136 [@bartelemi](https://github.com/bartelemi)
fix(list): 修复 ProFormList 使用 className 无效问题(#7128). [#7128](https://github.com/ant-design/pro-components/pull/#7128) [@leshalv](https://github.com/leshalv)
fix(list): onChange prop don't work ([#7129](https://github.com/ant-design/pro-components/pull/#7129)). #7129 [@kungege](https://github.com/kungege)
fix(table): Edit to make defaultValue become higher priority to reset column setting (#7113). [#7113](https://github.com/ant-design/pro-components/pull/#7113) [@ianccy](https://github.com/ianccy)
feat(table): use @dnd-kit (#7131). [#7131](https://github.com/ant-design/pro-components/pull/#7131) [@chenshuai2144](https://github.com/chenshuai2144)
fix(table): component EditableProTable actionDeleteRef function Single row status Locked (#7124). [#7124](https://github.com/ant-design/pro-components/pull/#7124) [@varown](https://github.com/varown)

**Comment by wangzhoubin** [comment]
> > @JackRoy1992 2.5.0修了
> > 没看到啊？？？
> > @ant-design/pro-components@2.5.0
> > 2023-05-29
> 
> fix(layout): 修复 collapsedShowTitle 样式错误问题 (#7098). [#7098](https://github.com/ant-design/pro-components/pull/#7098) [@ONLY-yours](https://github.com/ONLY-yours) fix(form): 解决 ProFormList 多重嵌套 ProFormList 下的表单组件未调用 transform 方法问题 (#7138). [#7138](https://github.com/ant-design/pro-components/pull/#7138) [@kaneruan](https://github.com/kaneruan) fix(form): use oninit reset formRef. [1deedf2](https://github.com/ant-design/pro-components/commit/1deedf2) fix(form): close icon style (#7091). [#7091](https://github.com/ant-design/pro-components/pull/#7091) [@kungege](https://github.com/kungege) fix(list): Extend ProListMeta type with renderFormItem ([#7136](https://github.com/ant-design/pro-components/pull/#7136)). #7136 [@bartelemi](https://github.com/bartelemi) fix(list): 修复 ProFormList 使用 className 无效问题(#7128). [#7128](https://github.com/ant-design/pro-components/pull/#7128) [@leshalv](https://github.com/leshalv) fix(list): onChange prop don't work ([#7129](https://github.com/ant-design/pro-components/pull/#7129)). #7129 [@kungege](https://github.com/kungege) fix(table): Edit to make defaultValue become higher priority to reset column setting (#7113). [#7113](https://github.com/ant-design/pro-components/pull/#7113) [@ianccy](https://github.com/ianccy) feat(table): use @dnd-kit (#7131). [#7131](https://github.com/ant-design/pro-components/pull/#7131) [@chenshuai2144](https://github.com/chenshuai2144) fix(table): component EditableProTable actionDeleteRef function Single row status Locked (#7124). [#7124](https://github.com/ant-design/pro-components/pull/#7124) [@varown](https://github.com/varown)

但是我最近用的还是有这个问题

**Comment by ldwonday** [comment]
@chenshuai2144 这个还是不行的哈。
https://codesandbox.io/p/sandbox/schema-biao-dan-forked-w883zf
